rfq/order.go

```diff
@@ -125,13 +127,20 @@ type AssetSalePolicy struct {
 	// wants us to produce NoOp HTLCs.
 	NoOpHTLCs bool
 
+	auxChanNegotiator *tapfeatures.AuxChannelNegotiator
```

Comment:
Could we replace the NoOpHTLCs flag and the auxChanNegotiator reference here with a single 'UseNoOpHTLCs' flag?

The ultimate purpose of including the auxChanNegotiator reference here is to eventually check if the peer supports NoOpHTLCs, which is then checked against our own NoOpHTLCs flag. But it would be nice to get away with storing only a single flag here, if that check could be resolved earlier. Could it be done in RegisterAssetSalePolicy or similar?

Reply:
(The further back it can be pushed the better. Possible it's necessary here as-is, but it's not immediately clear as to why.)